### ISSUE TITLE
[CI] Add Windows runner for automated testing

### ISSUE THREAD

**Issue by gruenich** [xiaoyeli/superlu#132]
We are testing Ubuntu to build and test SuperLU for every pull request. We should add a Windows runner to detect issues that only occur on Windows.

**Comment by gruenich** [comment]
#133 provides much of required code changes, but it does currently fail compilation without an error message.

**Comment by wo80** [comment]
Of course there's an error message:
```
D:\a\superlu\superlu\SRC\util.c(482,19): error C2057: expected constant expression [D:\a\superlu\superlu\build\SRC\superlu.vcxproj]
D:\a\superlu\superlu\SRC\util.c(482,25): error C2466: cannot allocate an array of constant size 0 [D:\a\superlu\superlu\build\SRC\superlu.vcxproj]
D:\a\superlu\superlu\SRC\util.c(482,12): error C2133: 'bucket': unknown size [D:\a\superlu\superlu\build\SRC\superlu.vcxproj]
```
And if you had read my comment https://github.com/xiaoyeli/superlu/pull/112#issuecomment-1823113141 carefully, you would know. This doe not work with VC compiler, so revert it: https://github.com/xiaoyeli/superlu/commit/e87529fd187937e43114fa2d3d15d74e24fe04c1

BTW: thank you for finally addressing this.

**Comment by wo80** [comment]
Adding the WinGetOpt dependency to actually run the tests on Windows shouldn't be hard, too. Find inspiration here: https://github.com/wo80/vs-arpack/blob/593ae85a26fbccd9e000cc32acdf55def643ea1b/.github/workflows/build.yml#L70-L79

Of course, you'll have to replace SuperLU with the WinGetOpt repo. You might also have to play around with the install prefix.

**Comment by wo80** [comment]
The fact that you overlooked the error message is probably due to an abundance of warnings given by the VC compiler. So while you're at it, you might at least disable the pretty useless C4996. Just add `_CRT_SECURE_NO_WARNINGS` here
https://github.com/xiaoyeli/superlu/blob/8f4f993e3af68d82de8fd0961b96abfacdf46f6e/CMakeLists.txt#L121-L123

**Comment by gruenich** [comment]
Thanks for the hint! I will include this, but I want to investigate whether fixing some of the warnings, e.g., the ones with insecure print functions.

**Comment by gruenich** [comment]
Fixed by #133 and #138.
